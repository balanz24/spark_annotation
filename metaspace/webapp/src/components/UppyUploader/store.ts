// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
