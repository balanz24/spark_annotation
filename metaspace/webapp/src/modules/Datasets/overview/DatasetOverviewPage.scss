$className: 'dataset-overview';
$img-viewer-bg-color: #F6F6F6;
$header-height: 64px;
$opt-desc-color: #C4C4C4;


.dataset-overview-container{
  @apply flex flex-row flex-wrap p-0;
  min-height: calc(100vh - #{$header-height});

  p {
    @apply my-0 max-w-measure-4;
  }

  h1 {
    @apply text-4xl leading-12 font-display;
  }
}


.dataset-overview-wrapper{
  @apply p-18 max-w-4xl box-border mx-auto leading-6 text-base;
}


.dataset-overview-img-wrapper{
  background: $img-viewer-bg-color;
  max-width: 100%;
}

.dataset-overview-holder{
  @apply flex flex-col justify-center px-6;
}


.dataset-overview-header{
  @apply  flex flex-row items-center justify-between px-6;

  .el-badge__content.is-fixed{
    top: 0px;
    right: 5px;
  }
}



.dataset-opt-description{
  @apply flex flex-row flex-wrap items-center border-solid justify-center border rounded p-2 my-4;
  border-color: #{$opt-desc-color};
  color: #{$opt-desc-color};
}


.dataset-overview-menu{
  z-index: 2000 !important;
}
