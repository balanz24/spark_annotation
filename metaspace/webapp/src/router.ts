/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
