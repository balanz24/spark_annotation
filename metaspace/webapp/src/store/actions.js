import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
