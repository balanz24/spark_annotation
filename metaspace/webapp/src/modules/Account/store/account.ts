import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
