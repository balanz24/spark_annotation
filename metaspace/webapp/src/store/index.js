import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
