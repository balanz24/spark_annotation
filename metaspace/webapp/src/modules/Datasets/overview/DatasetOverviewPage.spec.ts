import { mount } from '@vue/test-utils'
import DatasetOverviewPage from './DatasetOverviewPage'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import store from '../../../store'
import router from '../../../router'
import { initMockGraphqlClient, apolloProvider } from '../../../../tests/utils/mockGraphqlClient'

jest.mock('./DatasetActionsDropdown', () => ({ default: jest.fn() }))
jest.mock('../../../components/NewFeatureBadge', () => ({ default: jest.fn() }))

describe('DatasetOverviewPage', () => {
  const mockDataset = {
    id: 'dataset1',
    name: 'JD_Sampe',
    submitter: {
      id: 'userid',
      name: 'John Doe',
      email: 'jdoe@test.com',
    },
    principalInvestigator: {
      name: 'Test',
      email: 'jdoe@test.com',
    },
    group: {
      id: 'gxxxx',
      name: 'TEST_GROUP',
      shortName: 'TG',
    },
    groupApproved: true,
    projects: [
      {
        id: 'pj1',
        name: 'Test',
        publicationStatus: 'UNPUBLISHED',
      },
    ],
    isPublic: true,
    status: 'FINISHED',
    statusUpdateDT: '2021-03-22T01:54:04.856Z',
    metadataType: 'Imaging MS',
    canEdit: true,
    canDelete: true,
    canDownload: true,
    uploadDT: '2021-03-11T14:29:21.641Z',
    metadataJson: JSON.stringify({
      Metadata_Type: 'Imaging MS',
      MS_Analysis: {
        Detector_Resolving_Power: { mz: 1234, Resolving_Power: 123456 },
      },
    }),
    configJson: JSON.stringify({
      database_ids: [
        1,
      ],
      analysis_version: 3,
      isotope_generation: {
        adducts: [
          '+H',
        ],
        charge: 1,
        isocalc_sigma: 0.001238,
        instrument: 'Orbitrap',
        n_peaks: 4,
        neutral_losses: [],
        chem_mods: [],
      },
      fdr: {
        decoy_sample_size: 20,
        scoring_model: 'v3_default',
      },
      image_generation: {
        ppm: 3,
        n_levels: 30,
        min_px: 1,
        compute_unused_metrics: false,
      },
    }),
    description: JSON.stringify({
      type: 'doc',
      content: [
        {
          type: 'paragraph',
          content: [
            {
              type: 'text',
              text: 'Dataset 1.',
            },
          ],
        },
      ],
    }),
  }

  const testHarness = Vue.extend({
    components: {
      DatasetOverviewPage,
    },
    render(h) {
      return h(DatasetOverviewPage, { props: this.$attrs })
    },
  })

  const noDatasetQuery = () => {
    initMockGraphqlClient({
      Query: () => ({
        dataset: () => null,
        currentUser: () => ({ id: 'userid', role: 'user' }),
      }),
    })
  }
  const overviewQuery = () => {
    initMockGraphqlClient({
      Query: () => ({
        dataset: () => mockDataset,
        currentUser: () => ({ id: 'userid', role: 'user' }),
      }),
    })
  }

  beforeAll(() => {
    Vue.use(Vuex)
    sync(store, router)
    router.replace({
      name: 'dataset-overview',
      params: {
        dataset_id: 'dataset1',
      },
    })
  })

  it('should match snapshot when dataset exist', async() => {
    overviewQuery()
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })

  it('should match snapshot when dataset does not exist', async() => {
    noDatasetQuery()
    const wrapper = mount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })
})
