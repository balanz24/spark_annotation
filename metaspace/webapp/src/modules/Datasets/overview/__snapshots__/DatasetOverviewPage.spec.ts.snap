// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetOverviewPage should match snapshot when dataset does not exist 1`] = `<div class="text-center">This dataset doesn't exist, or you do not have access to it.</div>`;

exports[`DatasetOverviewPage should match snapshot when dataset exist 1`] = `
<div class="dataset-overview-container justify-center">
  <div class="dataset-overview-wrapper max-w-4xl w-full">
    <div class="dataset-overview-header">
      <h1 class="text-center truncate">JD_Sampe<span class="text-base align-middle"></span></h1>
      <!--function (el) {
      var obj;

      var args = [], len = arguments.length - 1;
      while ( len-- > 0 ) args[ len ] = arguments[ len + 1 ];
      if (shouldNotBeStubbed(el, stubs, modifiedComponents)) {
        return originalCreateElement.apply(void 0, [ el ].concat( args ))
      }

      if (isConstructor(el) || isComponentOptions(el)) {
        if (stubAllComponents) {
          var stub = createStubFromComponent(el, el.name || 'anonymous', _Vue);
          return originalCreateElement.apply(void 0, [ stub ].concat( args ))
        }
        var Constructor = shouldExtend(el) ? extend(el, _Vue) : el;

        return originalCreateElement.apply(void 0, [ Constructor ].concat( args ))
      }

      if (typeof el === 'string') {
        var original = resolveComponent(el, originalComponents);

        if (!original) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        if (isDynamicComponent(original)) {
          return originalCreateElement.apply(void 0, [ el ].concat( args ))
        }

        var stub$1 = createStubIfNeeded(stubAllComponents, original, _Vue, el);

        if (stub$1) {
          Object.assign(vm.$options.components, ( obj = {}, obj[el] = stub$1, obj ));
          modifiedComponents.add(el);
        }
      }

      return originalCreateElement.apply(void 0, [ el ].concat( args ))
    }-->
    </div>
    <div class="dataset-overview-holder">
      <p class="truncate">John Doe<a href="/group/gxxxx" class="ml-1">(TG)</a></p>
      <div>11 March, 2021</div>
      <section class="sm-RichText dataset-opt-description p-0">
        <div>
          <div class="transition-colors ease-in-out duration-300 rounded bg-transparent">
            <div contenteditable="false" tabindex="0" class="ProseMirror">
              <p data-empty-text=" " class="sm-RichText-placeholder is-editor-empty"><br></p>
            </div>
          </div>
        </div>
      </section>
    </div>
    <div class="dataset-overview-holder">
      <h1 class="truncate">Annotations</h1>
      <div class="relative text-center">
        <h4># of annotations with FDR% &lt;=</h4>
        <div class="el-table el-table--fit el-table--scrollable-x el-table--enable-row-hover el-table--enable-row-transition" style="width: 100%; margin-top: 20px;">
          <div class="hidden-columns">
            <div></div>
            <div></div>
            <div></div>
            <div></div>
            <div></div>
          </div>
          <div class="el-table__header-wrapper">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__header" style="width: 400px;">
              <colgroup>
                <col name="el-table_1_column_1" width="80">
                <col name="el-table_1_column_2" width="80">
                <col name="el-table_1_column_3" width="80">
                <col name="el-table_1_column_4" width="80">
                <col name="el-table_1_column_5" width="80">
              </colgroup>
              <thead class="">
                <tr class="">
                  <th colspan="1" rowspan="1" class="el-table_1_column_1     is-leaf is-sortable">
                    <div class="cell">Database<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_2     is-leaf is-sortable">
                    <div class="cell">5%<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_3     is-leaf is-sortable">
                    <div class="cell">10%<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_4     is-leaf is-sortable">
                    <div class="cell">20%<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                  <th colspan="1" rowspan="1" class="el-table_1_column_5     is-leaf is-sortable">
                    <div class="cell">50%<span class="caret-wrapper"><i class="sort-caret ascending"></i><i class="sort-caret descending"></i></span></div>
                  </th>
                </tr>
              </thead>
            </table>
          </div>
          <div class="el-table__body-wrapper is-scrolling-left">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__body" style="width: 400px;">
              <colgroup>
                <col name="el-table_1_column_1" width="80">
                <col name="el-table_1_column_2" width="80">
                <col name="el-table_1_column_3" width="80">
                <col name="el-table_1_column_4" width="80">
                <col name="el-table_1_column_5" width="80">
              </colgroup>
              <tbody>
                <tr class="el-table__row">
                  <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=115&amp;ds=dataset1" class="">dataset.annotationCounts.0.dbName-dataset.annotationCounts.0.dbVersion</a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=115&amp;ds=dataset1&amp;fdr=0.05" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=115&amp;ds=dataset1&amp;fdr=0.1" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_4  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=115&amp;ds=dataset1&amp;fdr=0.2" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=115&amp;ds=dataset1&amp;fdr=0.5" class=""></a></div>
                  </td>
                </tr>
                <tr class="el-table__row">
                  <td rowspan="1" colspan="1" class="el-table_1_column_1  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=114&amp;ds=dataset1" class="">dataset.annotationCounts.1.dbName-dataset.annotationCounts.1.dbVersion</a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_2  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=114&amp;ds=dataset1&amp;fdr=0.05" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_3  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=114&amp;ds=dataset1&amp;fdr=0.1" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_4  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=114&amp;ds=dataset1&amp;fdr=0.2" class=""></a></div>
                  </td>
                  <td rowspan="1" colspan="1" class="el-table_1_column_5  ">
                    <div class="cell"><a href="/dataset/dataset1/annotations?db_id=114&amp;ds=dataset1&amp;fdr=0.5" class=""></a></div>
                  </td>
                </tr>
                <!---->
              </tbody>
            </table>
            <!---->
            <!---->
          </div>
          <div class="el-table__footer-wrapper">
            <table cellspacing="0" cellpadding="0" border="0" class="el-table__footer" style="width: 400px;">
              <colgroup>
                <col name="el-table_1_column_1" width="80">
                <col name="el-table_1_column_2" width="80">
                <col name="el-table_1_column_3" width="80">
                <col name="el-table_1_column_4" width="80">
                <col name="el-table_1_column_5" width="80">
              </colgroup>
              <tbody class="">
                <tr>
                  <td colspan="1" rowspan="1" class="el-table_1_column_1 is-leaf">
                    <div class="cell">Total Annotations</div>
                  </td>
                  <td colspan="1" rowspan="1" class="el-table_1_column_2 is-leaf">
                    <div class="cell"><a href="/dataset/dataset1/annotations?ds=dataset1&amp;fdr=0.05" class="">NaN</a></div>
                  </td>
                  <td colspan="1" rowspan="1" class="el-table_1_column_3 is-leaf">
                    <div class="cell"><a href="/dataset/dataset1/annotations?ds=dataset1&amp;fdr=0.1" class="">NaN</a></div>
                  </td>
                  <td colspan="1" rowspan="1" class="el-table_1_column_4 is-leaf">
                    <div class="cell"><a href="/dataset/dataset1/annotations?ds=dataset1&amp;fdr=0.2" class="">NaN</a></div>
                  </td>
                  <td colspan="1" rowspan="1" class="el-table_1_column_5 is-leaf">
                    <div class="cell"><a href="/dataset/dataset1/annotations?ds=dataset1&amp;fdr=0.5" class="">NaN</a></div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <!---->
          <!---->
          <!---->
          <div class="el-table__column-resize-proxy" style="display: none;"></div>
        </div>
        <div class="text-right mt-2"><button type="button" class="el-button el-button--default">
            <!---->
            <!----><span>Browse annotations</span></button></div>
      </div>
    </div>
    <div class="dataset-overview-holder">
      <h1 class="truncate">Details</h1>
      <div class="flex flex-wrap relative -m-3">
        <div class="flex-grow box-border min-w-64 p-3 break-words">
          <p class="max-h-32 overflow-y-auto">Imaging MS</p>
        </div>
        <div class="flex-grow box-border min-w-64 p-3 break-words">
          <h3 class="m-0 my-2">MS analysis</h3>
          <ul class="list-none p-0 m-0 max-h-40 overflow-y-auto"></ul>
        </div>
      </div>
    </div>
    <div class="dataset-overview-holder">
      <div class="flex flex-wrap relative -m-3">
        <div class="flex-grow box-border min-w-64 p-3 break-words">
          <h3 class="m-0 my-2">Annotation settings</h3>
          <ul class="list-none p-0 m-0 max-h-40 overflow-y-auto">
            <li><b>m/z tolerance (ppm):</b> 3</li>
          </ul>
        </div>
      </div>
    </div>
    <div class="dataset-overview-holder">
      <h1 class="truncate">Projects</h1>
      <div class="flex-grow box-border min-w-64 p-0 break-words">
        <ul class="list-none p-0 py-3 m-0 max-h-40 overflow-y-auto">
          <li><b>Name: </b><a href="/project/pj1" class="ml-1">Test</a></li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
